### ISSUE TITLE
O_DIRECTORY not defined on sparc solaris 5.10

### ISSUE THREAD

**Issue by gambiteer** [gambit/gambit#895]
I'm trying to build github mainline on
```
bash-5.2$ uname -a
SunOS arthur.cs.purdue.edu 5.10 Generic_150400-59 sun4v sparc SUNW,Sun-Fire-T200 Solaris
```
(Don't ask.)  While building `gsc-boot` it fails with
```
os_files.c: In function 'at_long_path':
os_files.c:543:53: error: 'O_DIRECTORY' undeclared (first use in this function)
           new_dir = ___openat_no_EINTR (dir, start, O_DIRECTORY, 0);
                                                     ^
os_files.c:543:53: note: each undeclared identifier is reported only once for each function it appears in
os_files.c: In function 'opendir_long_path':
os_files.c:963:48: error: 'O_DIRECTORY' undeclared (first use in this function)
       int fd = ___openat_no_EINTR (dir, path2, O_DIRECTORY, 0);
                                                ^
make[1]: *** [makefile:355: os_files.o] Error 1
```
Is there a short hack I can make to the 4.9.5 sources to fix this?

**Comment by feeley** [comment]
Could you try this: https://stackoverflow.com/questions/53693398/c-language-error-o-directory-undeclared-and-accessing-files-in-directory-and

**Comment by gambiteer** [comment]
No go.  I tried
```
bash-5.2$ ./configure CC='gcc -D_POSIX_C_SOURCE=200809L'
```
got lots of messages like:
```
configure: WARNING: sys/resource.h: present but cannot be compiled
configure: WARNING: sys/resource.h:     check for missing prerequisite headers?
configure: WARNING: sys/resource.h: see the Autoconf documentation
configure: WARNING: sys/resource.h:     section "Present But Cannot Be Compiled"
configure: WARNING: sys/resource.h: proceeding with the compiler's result
configure: WARNING:     ## -------------------------------------- ##
configure: WARNING:     ## Report this to gambit@iro.umontreal.ca ##
configure: WARNING:     ## -------------------------------------- ##
```
and many compile time errors.  The same thing happened with
```
bash-5.2$ ./configure CC='gcc -D_XOPEN_SOURCE=700'
```
With
```
bash-5.2$ ./configure CC='gcc -D_GNU_SOURCE'            
```
I again get
```
os_files.c: In function 'at_long_path':
os_files.c:543:53: error: 'O_DIRECTORY' undeclared (first use in this function)
           new_dir = ___openat_no_EINTR (dir, start, O_DIRECTORY, 0);
                                                     ^
os_files.c:543:53: note: each undeclared identifier is reported only once for each function it appears in
os_files.c: In function 'opendir_long_path':
os_files.c:963:48: error: 'O_DIRECTORY' undeclared (first use in this function)
       int fd = ___openat_no_EINTR (dir, path2, O_DIRECTORY, 0);
                                                ^
```
FWIW, `config.guess` gives
```
checking build system type... sparc-sun-solaris2.10
checking host system type... sparc-sun-solaris2.10
checking target system type... sparc-sun-solaris2.10
```

**Comment by feeley** [comment]
This might be the "correct" solution: https://lists.nongnu.org/archive/html/bug-mailutils/2021-06/msg00008.html

**Comment by gambiteer** [comment]
That keeps the build going.  I get a couple warnings:
```
os_base.c: In function '___free_mem_code':
os_base.c:405:11: warning: passing argument 1 of 'munmap' from incompatible pointer type
   munmap (p, *p + sizeof (___SIZE_TS));
           ^
In file included from /usr/include/dlfcn.h:18:0,
                 from os.h:1383,
                 from os_base.h:8,
                 from os_base.c:13:
/usr/include/sys/mman.h:169:12: note: expected 'caddr_t' but argument is of type 'int *'
 extern int munmap(caddr_t, size_t);
            ^
```

**Comment by gambiteer** [comment]
I'm going to close this as "Won't fix".
